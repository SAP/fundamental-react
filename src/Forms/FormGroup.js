import classnamesBind from 'classnames/bind';
import PropTypes from 'prop-types';
import React from 'react';
import withStyles from '../utils/withStyles';
import styles from 'fundamental-styles/dist/form-group.css';

const classnames = classnamesBind.bind(styles);

/** A **FormGroup** encapsulates components related to a single form input.
 * At a minimum, it should contain **FormLabel** and input child components. */
const FormGroup = React.forwardRef(({ children, cssNamespace, ...props }, ref) => {

    // Note that we don't want to include the fd-form-group--inline
    // class here because it is only for FormRadioGroup

    return (<div {...props} className={classnames(`${cssNamespace}-form-group`)}
        ref={ref}>{children}</div>);
});

FormGroup.displayName = 'FormGroup';

FormGroup.propTypes = {
    /** Node(s) to render within the component */
    children: PropTypes.node
};

export default withStyles(FormGroup);
