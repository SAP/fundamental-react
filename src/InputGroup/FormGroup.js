import PropTypes from 'prop-types';
import React from 'react';
import withStyles from '../utils/WithStyles/WithStyles';

const FormGroup = React.forwardRef(({ children, disableStyles, ...props }, ref) => {
    return (<div {...props} className='fd-form-group'
        ref={ref}>{children}</div>);
});

FormGroup.displayName = 'FormGroup';

FormGroup.propTypes = {
    children: PropTypes.node,
    customStyles: PropTypes.object,
    disableStyles: PropTypes.bool
};

export { FormGroup as __FormGroup };

export default withStyles(FormGroup, { cssFile: 'form-group' });
