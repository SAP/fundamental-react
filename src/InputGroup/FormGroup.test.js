import FormGroup from './FormGroup';
import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

describe('<FormGroup />', () => {
    const formGroup = <FormGroup>Test Form Group</FormGroup>;

    test('create snapshot', () => {
        // create form group
        let component = renderer.create(formGroup);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe('Prop spreading', () => {
        test('should allow props to be spread to the FormGroup component', () => {
            const element = mount(<FormGroup data-sample='Sample' />);

            expect(
                element.getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });
    });
    test('forwards the ref', () => {
        let ref;
        class Test extends React.Component {
            constructor(props) {
                super(props);
                ref = React.createRef();
            }
            render = () => <FormGroup ref={ref} />;
        }
        mount(<Test />);
        expect(ref.current.tagName).toEqual('DIV');
        expect(ref.current.className).toEqual('fd-form-group');
    });
});
