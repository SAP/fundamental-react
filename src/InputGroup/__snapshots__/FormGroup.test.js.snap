// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormGroup /> create snapshot 1`] = `
<div
  className="fd-form-group"
>
  Test Form Group
</div>
`;
